File: phalcon/app/model/UserInfo.php
<?php
namespace app\model;

/* 用户信息表 */
class UserInfo extends Model {

  public $uid;
  protected $birthday;
  protected $utime;
  
  /* 构造函数 */
  public function initialize(){
    $this->setSource('user_info');  //数据表
  }

  /* 生日 */
  function setBirthday($val){
    if(empty($val)){
      $val = null;
    }
    $this->birthday = $val;
  }
  function getBirthday(){
    return $this->birthday;
  }

  /* 更新 */
  public function beforeUpdate(){
    $this->utime = date('YmdHis');
  }

}

File: java/src/main/java/vip/webmis/java/model/UserInfo.java
package vip.webmis.java.model;

import com.alibaba.fastjson.JSONObject;

import vip.webmis.java.common.Inc;

/* 用户信息表 */
public class UserInfo extends Model {

  public String uid = "0";
  public String utime = "null";
  public String nickname = "";
  public String position = "";
  public String name = "";
  public String gender = "";
  public String birthday = "null";
  public String img = "";
  
  /* 构造函数 */
  public UserInfo() {
    this.setPrimaryKey("uid"); //主键
    this.setSource("user_info"); //数据表
  }

  /* 生日 */
  public void setBirthday(String val) throws Exception {
    if(val.equals("")){
      val = "null";
    }
    birthday = val;
  }
  public String getBirthday(){
    return birthday;
  }

  /* 更新 */
  public JSONObject beforeUpdate(JSONObject fields){
    if(fields.get("utime").equals("null")){
      fields.put("utime",Inc.date("yyyy-MM-dd HH:mm:ss"));
    }
    return fields;
  }

}
